kover-gradle-plugin/docs/index.md

```diff
@@ -36,11 +42,713 @@ Gradle plugin that measures the coverage for tests running on the JVM and genera
+You [do not need to specify](#application-in-submodule-with-plugins-dsl) the plugin version in all submodules.
+
+For more information about application of the plugin, refer to the [relevant section](#apply-kover-gradle-plugin-in-project)
+
+After applying Kover Gradle plugin, the classes are modified (instrumented) when loaded into the JVM during local unit test runs,
+so number of line hits can be counted. This may impact your testing experience.
+At the moment, the connected tests are not being instrumented.
+
+To learn more about instrumentation, it's configuring, and solutions for typical problems, please refer to the [corresponded section](#instrumentation).
+
+
+
+#### Using Android Build Variants
+In the Android Gradle Plugin, the module is divided into different **build variants** (flavor + build type), which may have different source sets and tests.
+For each Gradle Build Variant, Kover creates its own **report variant** that have the same names.
+
+The **report variant** includes only the classes declared in the corresponding **build variant**, and also runs only tests this **build variant**.
+
+If there is a need to generate report from classes of several **build variants**, then you could create custom **report variant** in each module:
+```kotlin
+kover {
+    currentProject {
+        createVariant("custom") {
+            add("libDebug")
+            add("appDebug")
+        }  
+    }
+}
+```
+Thus, a **report variant** `custom` will be created, which contains classes and tests from the `libDebug` and `appDebug` **build variants**.
+
+#### Total report variant
+Kover also creates a special **report variant**, which includes all classes and all tests of the module. This variant is called the **total variant** - and it does not have a special name.
+The names of the **report variants** are used when configuring the report and when running report generation tasks.
+
+#### Different Build Variants in different modules
+Suppose you have two modules, `:app1`, `:app2` and build variants differ between these unrelated modules.
+Let it be `debug` and `release` in the `:app1`, and `mainDebug`, `extraDebug` and `release` in the `:app2`.
+
+If you need to create a report containing classes from these two modules at the same time, 
+but a limited set of build variants, then you should create a custom report variant in each module.
+
+For example, you need classes from `debug` build variant from `:app1`, and `extraDebug` from `:app2`.
+Then you need to add these variants to the custom one in the appropriate modules:
+
+In `:app1`
+```kotlin
+kover {
+    currentProject {
+        createVariant("custom") {
+            add("debug")
+        }  
+    }
+}
+```
+And in `:app2`
+```kotlin
+kover {
+    currentProject {
+        createVariant("custom") {
+            add("extraDebug")
+        }  
+    }
+}
+```
+
+After that, you should use the report generation tasks only for this custom variant, e.g. `koverHtmlReportCustom`.
```

Comment:
Do I get 1 report out of it? Does it work even if the merging module doesn't declare `custom` explicitly?

Reply:
It is expected that it will be created in all modules, this is described above:
`you should create a custom report variant in each module.`